client/js/lounge.js

```diff
@@ -369,7 +396,7 @@ $(function() {
 
 	function renderChannelMessages(data) {
 		var documentFragment = buildChannelMessages(data.id, data.messages);
-		var channel = chat.find("#chan-" + data.id + " .messages").append(documentFragment);
+		var channel = chat.find("#chan-" + data.id + " .messages").html(documentFragment);
```

Comment:
On normal load `#chan-" + data.id + " .messages` is always empty, so it's fine. On reconnection this block should be re-rendered, thus `append` was replaced.